### DISCUSSION TITLE
RS232/TTL failures

### DISCUSSION THREAD

**Discussion by wrvolz** [jgyates/genmon#1125] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

Latest

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Occasionally when there is a spike on the utility power the RS232/TTL interface board will stop communication with the generator. Replaceing the board fixes the problem. I've replaced at least 5 of these and more are ordered and the current one failed. The RPI is grounded to the battery. The RPI is powered by POE with a splitter to the USB power plug.  What can I do to prevent this? Would powering the RPI from the battery instead help?

**Comment by skipfire** [comment]
Powering from the battery may help, or a higher quality TTL board.  From the setup you described, having a board with an optoisolator may help as well.

**Reply by skipfire** [reply]
Also how do you have the TTL board powered?  5v from the Molex, 3v3 from the Pi, 5v from the Pi, or something else?

**Reply by wrvolz** [reply]
It looks like I'm using GPIO pins 4,6,8,10 so 5V. The TTL board is one recommended by this site, HiLetGo.

**Reply by skipfire** [reply]
There have been other reports of the same issue with the boards failing.  You may want to try powering it by 3v3 instead.  That should be healthier for the Pi anyway as while 5v tends to not damage the Pi, Pi is not listed as 5v tolerant.
